src/client/scripts/views/SignedIn/DiscoverView.js

```diff
@@ -131,35 +131,97 @@ export class DiscoverView {
-            // curUser has not already liked, rejected, or matched with user
-            !this.#curUser.liked.includes(user.id)       &&
-            !this.#curUser.rejected.includes(user.id)    &&
-            !this.#curUser.matches.includes(user.id);
-
-        const fitsPreferences = (user) =>
-            this.#curUser.hasHousing !== user.hasHousing &&
-            user.id !== this.#curUser.id && 
-            this.#curUser.preferences.cities.includes(user.housing.city) &&
-            user.housing.rent <= this.#curUser.preferences.rent.max &&
-            this.#curUser.preferences.gender[user.housing.gender] &&
-            this.#curUser.preferences.leaseLength[user.housing.leaseLength] &&
-            this.#curUser.preferences.leaseType[user.housing.leaseType] &&
-            this.#curUser.preferences.roomType[user.housing.roomType] &&
-            this.#curUser.preferences.buildingType[user.housing.buildingType] &&
-            this.#curUser.preferences.timeframe[user.housing.timeframe] &&
-             Object.keys(this.#curUser.preferences.amenities).some(
+  /**
+ * Gets an array of users that the current user hasn't liked, rejected, or matched with yet,
+ * and whose housing preferences match the current user's preferences.
+ *
+ * @returns {Promise<User[]>} A promise that resolves with an array of user objects.
+ */
+async #getUnseenUsersWithoutHousing() {
+    const allUsers = await getAllUsers();
+
+    const fitsRequirements = (user) =>
+        // user has housing if curUser doesn't, vice versa
+        this.#curUser.hasHousing !== user.hasHousing &&
+        // user is not curUser
+        user.id !== this.#curUser.id &&
+        // curUser has not already liked, rejected, or matched with user
+        !this.#curUser.liked.includes(user.id) &&
+        !this.#curUser.rejected.includes(user.id) &&
+        !this.#curUser.matches.includes(user.id);
+
+    const fitsPreferences = (user) =>
+        this.#curUser.hasHousing !== user.hasHousing &&
+        !this.#curUser.hasHousing &&
+        user.id !== this.#curUser.id &&
+        this.#curUser.preferences.cities.includes(user.housing.city) &&
+        user.housing.rent <= this.#curUser.preferences.rent.max &&
+        this.#curUser.preferences.gender[user.housing.gender] &&
+        this.#curUser.preferences.leaseLength[user.housing.leaseLength] &&
+        this.#curUser.preferences.leaseType[user.housing.leaseType] &&
+        this.#curUser.preferences.roomType[user.housing.roomType] &&
+        this.#curUser.preferences.buildingType[user.housing.buildingType] &&
+        this.#curUser.preferences.timeframe[user.housing.timeframe] &&
+        Object.keys(this.#curUser.preferences.amenities).some(
             amenity => this.#curUser.preferences.amenities[amenity] && user.housing.amenities.includes(amenity)
         );
-        return allUsers.filter(fitsRequirements).filter(fitsPreferences);
+
+    return allUsers.filter(fitsRequirements).filter(fitsPreferences);
```

Comment:
I'm thinking it might be easier to just make a comparison function, and call with different arguments depending on whether the user has housing. That way we can have just 1 `getUnseenUsers`:
```
 async #getUnseenUsers() {
    const allUsers = await getAllUsers();

    const fitsRequirements = (user) => {
        const unseen = this.#curUser.hasHousing !== user.hasHousing && 
            user.id !== this.#curUser.id &&
            !this.#curUser.liked.includes(user.id) &&
            !this.#curUser.rejected.includes(user.id) &&
            !this.#curUser.matches.includes(user.id);

        const compatible = this.#curUser.hasHousing
            ? this.#isCompatible(user.preferences, this.#curUser.housing)
            : this.#isCompatible(this.#curUser.preferences, user.housing);

        return unseen && compatible;
    }

    return allUsers.filter(fitsRequirements);
}

#isCompatible(prefs: Preferences, housing: Housing) {
    return prefs.cities.includes(housing.city) &&
        prefs.gender[housing.gender] &&
        // etc.
}

```

Reply:
oh whoops did not see you did all that work my b💀